lib/hive/port_allocator.rb

```diff
@@ -37,6 +37,8 @@ def allocate_port
 
     # Relase a single port in the range
     def release_port(p)
+      pid = `lsof -i | grep #{p} | awk '{print $2}'`.strip
+      Process.kill 0, pid if !pid.empty?
```

Comment:
Can there please be a warning entered in the log when a process needs to be killed. Ideally, the process should already be terminated by this point so a warning would flag up an issue in the worker that needs to be addressed.